docs: update Getting started docs

bajtos opened the pull request:
- chore: setup Prettier formatting
- feat: specify supported Node.js versions
- docs: update Getting started docs

Resolve #25

bajtos commented on package.json:
```diff
@@ -8,15 +8,19 @@
     "station": "./bin/station.js"
   },
+  "engines": {
+    "node": ">=18.12"
```
Hmm, I see that the CI is running on Node.js 16 too. Is there any particular reason why we should support that version line?

Either way, we should use the same minimal supported Node.js version in all three places - CI workflows, package.json, and documentation.

juliangruber replied:
Node 16 has EOL 2023-09-11, so is still a relevant and supported version (https://github.com/nodejs/Release). Or would you like to choose different criteria?

juliangruber replied:
Are you ok with supporting node 16? If yes, please adjust the other places to match

bajtos replied:
The fewer versions we support, the less maintenance work for us.

Considering that this is a CLI, not a library, I would assume people have more flexibility in choosing the Node.js version and thus be able to run on the latest LTS.

Anyhow, the goal of this PR is to add docs, so I'll update my changes to include support for Node.js 16.x.

bajtos replied:
Resolved in https://github.com/filecoin-station/core/pull/47/commits/5cb8a745a1c020e995bfffc2b7c12acc118f806b

juliangruber commented on README.md:
```diff
@@ -9,12 +9,23 @@
+
 ## Installation
 
+Station Core requires Node.js version 18.12 or newer. We recommend using the
```
see Node.js version discussion

bajtos replied:
Resolved in https://github.com/filecoin-station/core/pull/47/commits/5cb8a745a1c020e995bfffc2b7c12acc118f806b

juliangruber commented on README.md:
```diff
@@ -25,26 +36,40 @@ $ npm install -g @filecoin-station/core
 
 ### `$ station metrics`
 
+_This command reads information from the state files created by the main Station
```
"main Station process" is a term not yet established. Also I don't think we need to talk about files here, does the user care where that data is coming from? To me this is too much an implementation detail. Therefore I wonder what you're trying to achieve by adding this text 🤔

bajtos replied:
I am so glad you started this discussion, @juliangruber 🙏🏻

juliangruber replied:
Hmm, if you compare this to `docker`, the `logs` command doesn't care if the container is running or not. I agree though that we should prevent concurrent `$ station` calls!

To me, annotating that logs are outdated doesn't make sense. The logs won't be outdated, they will be as uptodate as they can be. Only nothing is running that can produce new logs.

Can you maybe think of another CLI that works the way you propose?

juliangruber replied:
> Maybe we can also improve all commands to print information about the location of state files at startup to make troubleshooting easier.

Are you sure this is a problem? Have you run into this?

juliangruber replied:
> If you don't have the main Station Core process running, there will be either no metrics and no logs, or the data will be outdated.

To me it's obvious that there will be no new metrics or logs if the thing isn't running. But maybe I have a blind spot? It's the same with systemd: You can always get the logs. If you don't get new logs, you check if the service is running, and otherwise start it.

juliangruber commented on README.md:
```diff
@@ -64,17 +89,18 @@ $ station metrics --follow
+process._
 
-Get all logs:
+Get a combined log from the Station Core and all Station Modules:
```
```suggestion
Get combined logs from Station Core and all Station Modules:
```

juliangruber commented on README.md:
````diff
@@ -9,14 +9,25 @@
+
 ```bash
-$ npm install -g @filecoin-station/core
+$ npm install -g --engine-strict @filecoin-station/core
````
Meh, that's ugly to me :| It should be the user's choice how they want to use NPM.

bajtos replied:
👍🏻

juliangruber commented on package.json:
```diff
@@ -8,15 +8,19 @@
     "station": "./bin/station.js"
   },
+  "engines": {
+    "node": "^16.13 || >=18.12"
```
why shouldn't someone on node 18.0 be able to run this? I don't see an advantage in forcing people to upgrade nodejs. Depending on the environment, they might not be able to (corporate policies etc), unnecessary limiting the possible install base of Station.
```suggestion
    "node": ">=16"
```

bajtos replied:
The 16.x versions before 16.13 and 18.x versions before 18.12 were not LTS yet.

In the past, it was necessary to choose a non-zero server-minor version to get access to features added later in the lifecycle of the major version. E.g. top-level await was added in 14.8.

Anyhow, I don't have any particular argument or requirement to reject older 16.x and 18.x, so I am fine to drop the semver-minor part from the version spec.

However: your suggested change adds support for Node.js 17, which is no longer maintained. Are you fine with that? We don't run CI for that version.

bajtos replied:
Also: do we want to automatically support all future Node.js versions by using `>=18`? Maybe the process of adding support for a new major version should be manual - add the new version to the CI matrix first, ensure all tests pass, only then update the version range.

bajtos commented on README.md:
```diff
@@ -9,12 +9,27 @@
+> **Note**: Station Core requires Node.js, we recommend using the latest LTS
+> version. You can install Node.js using your favorite package manager or get
+> the official installer from
+> [Node.js downloads](https://nodejs.org/en/download/).
```
Rendered version:

<img width="1045" alt="Screenshot 2023-03-15 at 14 36 13" src="https://user-images.githubusercontent.com/1140553/225325014-82d41a01-f9f1-4db2-8f22-9221cd75b772.png">

See https://github.com/community/community/discussions/16925

juliangruber replied:
This is looking awesome! 👏

bajtos commented:
Let's not bike shed this too much.

I think I addressed all your comments:
- Removed mentions of the main process.
- Removed specific Node.js versions from the README so that we don't have to remember to update README when we change which major versions are supported.
- Reworked engines spec to `>=16`.

Is there anything else needed before we can land this PR?

I would prefer to finish this ASAP, and then open follow-up pull requests if we want to make more changes.
